R/meta.R

```diff
@@ -66,20 +66,59 @@ dm_meta_raw <- function(con, catalog) {
     # "collation_catalog", "collation_schema", "domain_catalog",
     # "domain_schema", "domain_name"
   ))
+  
+  # add is_autoincrement column to columns table
+  if (is_mssql(src)) {
+    columns <- columns %>%
+      mutate(is_autoincrement = sql("CAST(COLUMNPROPERTY(object_id(TABLE_SCHEMA+'.'+TABLE_NAME), COLUMN_NAME, 'IsIdentity') AS BIT)"))
+  } else if (is_postgres(src)) {
+    if("info" %in% slotNames(src$con) && !is.null(src$con@info$db.version) && as.package_version(src$con@info$db.version) >= as.package_version("15.0")) {
+      columns <- columns %>%
+        mutate(is_autoincrement = sql("REGEXP_LIKE(column_default, 'nextval')"))
+    } else {
+      columns <- columns %>%
+        mutate(is_autoincrement = sql("REGEXP_MATCH(column_default, 'nextval')")) %>%
+        mutate(is_autoincrement = if_else(!is.na(is_autoincrement) & is_autoincrement == "{nextval}", TRUE, FALSE))
+    }
+    
+  } else {
```

Comment:
@krlmlr how do you feel about having different logic for different the db versions? REGEXP_MATCH function works on every postgres version, but is messy. REGEXP_LIKE function is only introduced in 15.0, but does the job perfectly. All interfaces don't report the db version as nicely as `odbc::odbc()` does, so that makes identifying it a bit harder.. I think we just use the REGEXP_MATCH approach and forget the versions for now.